src/v2.rs

```diff
@@ -482,6 +482,13 @@ impl EszipV2 {
     parser: &CapturingModuleParser,
     mut emit_options: EmitOptions,
   ) -> Result<Self, anyhow::Error> {
+    graph.valid()?;
+
+    // ensure there are no errors in any module slots (ex. redirects)
+    if let Some(error) = graph.errors().next() {
+      return Err(error.clone().into());
+    }
```

Comment:
It's better to check all this when creating the eszip because then it forces that it's done properly rather than leaving it up to the caller.